File: DSA_Level-1/Basics Of Programming/2D Arrays/The State Of Wakanda - 2/TheStateOfWakanda-2.js

function diagonalDisplay(nums){
    for(let j=0;j<nums.length;++j){
        for(let i=0;i<nums.length-j;++i){
            console.log(nums[i][i+j]+" ");
        }
    }
}

diagonalDisplay([[11,12,13,14],[21,22,23,24],[31,32,33,34],[41,42,43,44]]);

File: DSA_Level-1/Basics Of Programming/2D Arrays/The State Of Wakanda - 2/TheStateOfWakanda-2.java

import java.io.*;
import java.util.*;

public class Main {

  public static void main(String[] args) throws Exception {
    Scanner scn = new Scanner(System.in);
    int n = scn.nextInt();
    int[][] nums = new int[n][n];

    for (int i = 0; i < n; ++i) {
      for (int j = 0; j < n; ++j) {
        nums[i][j] = scn.nextInt();
      }
    }

    for (int j = 0; j < nums.length; ++j) {
      for (int i = 0; i < nums.length - j; ++i) {
        System.out.println(nums[i][i + j] + " ");
      }
    }
  }

}